### DISCUSSION TITLE
NC container in endless starting loop during setup

### DISCUSSION THREAD

**Discussion by p3palazzo** [nextcloud/all-in-one#3162] [Questions] [answered]
I am trying to set up an instance of Nextcloud AIO with `--env NEXTCLOUD_DATADIR="/mnt/ncdata"` on an Ubuntu 22 server that has its own subdomain name in addition to an IP address.

The first time I tried to set it up, it hung after I selected the addons, showing a "failed to talk to server" message on the top right-hand corner of the browser window. Now, every time I retry the installation, the Nextcloud container gets stuck in an endless starting loop with the following error message:

```
(1 row)

+ '[' -f /dev-dri-group-was-added ']'
++ find /dev -maxdepth 1 -mindepth 1 -name dri
+ '[' -n '' ']'
+ set +x
Configuring Redis as session handler...
Setting php max children...
Nextcloud is not installed - only a limited number of commands are available
Appdata is not present. Did you maybe change the datadir after the initial Nextcloud installation? This is not supported!
See https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
If you adjusted the datadir to be located on an external drive, make sure that the drive is still mounted!
In the datadir was found:
total 8
drwxr-x--- 2 www-data root     4096 Aug 15 00:57 .
drwxr-xr-x 1 root     root     4096 Aug 15 00:56 ..
-rw-r--r-- 1 www-data www-data    0 Aug 15 00:56 nextcloud.log
```

Now, I most definitely did NOT change the datadir at any time. The `--env` setting has been the same ever since my first failed installation attempt.

I tried deleting all the containers and images including the mastercontainer, removing the datadir, and re-running the docker command. The AIO mastercontainer always recreates the datadir, but it also looks like it is pulling information about an ongoing installation from somewhere else. Here is what the AIO logs say:

```
Trying to fix docker.sock permissions internally...
Adding internal www-data to group ping
[0;92mInitial startup of Nextcloud All-in-One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080

If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443[0m
[15-Aug-2023 03:48:31] NOTICE: fpm is running, pid 100
[Tue Aug 15 03:48:31.483606 2023] [mpm_event:notice] [pid 96:tid 140584313760584] AH00489: Apache/2.4.57 (Unix) OpenSSL/3.1.1 configured -- resuming normal operations
{"level":"info","ts":1692071311.48783,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
[15-Aug-2023 03:48:31] NOTICE: ready to handle connections
[Tue Aug 15 03:48:31.490658 2023] [core:notice] [pid 96:tid 140584313760584] AH00094: Command line: 'httpd -D FOREGROUND'
```

I tried both using the domain-name:8443 and ip-address:8080 with identical results. I would *really* like to scrap the whole installation and start fresh, but short of deleting my VM I cannot for the life of me find out how to do this. I delete everyting I can find about the AIO installer—docker containers and images and the datadir—but it always seems like it is reading some phantom data about the ongoing installation from somewhere else.

**Comment by szaimen** [comment, marked as answer]
Can you reset things properly and try again?

https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance

**Reply by p3palazzo** [reply]
Thanks! I missed this part of the README, and I had not realized how long it took for the Nextcloud container to spin up the first time so I broke things by not waiting long enough. Everything seems to be working now.

May I suggest that the "failed to talk to server" message that comes up when the containers are starting is somewhat misleading and not very informative? It was this that led me to try and restart the containers when I should really have waited things out.

**Reply by Corran2015** [reply]
Agreeing with p3palazzo; when it gave me the "failed to talk to server" error, I went looking for how to restart the thing. Thankfully, searching that quoted line brought me here.
